code/modules/paperwork/paper.dm

```diff
@@ -304,53 +305,81 @@
 		\[time\] : Inserts the current station time in HH:MM:SS.<br>
 	</BODY></HTML>"}, "window=paper_help")
 
+/obj/item/paper/proc/topic_href_write(id, input_element)
+	var/obj/item/item_write = usr.get_active_hand() // Check to see if he still got that darn pen, also check if he's using a crayon or pen.
+	add_hiddenprint(usr) // No more forging nasty documents as someone else, you jerks
+	if(!istype(item_write, /obj/item/pen) && !istype(item_write, /obj/item/toy/crayon))
+		return
+	if(src.loc != usr && !src.Adjacent(usr) && !((istype(src.loc, /obj/item/clipboard) || istype(src.loc, /obj/item/folder)) && (src.loc.loc == usr || src.loc.Adjacent(usr))))
+		return // If paper is not in usr, then it must be near them, or in a clipboard or folder, which must be in or near usr
+/*
+	t = checkhtml(t)
+	// check for exploits
+	for(var/bad in paper_blacklist)
+		if(findtext(t,bad))
+			to_chat(usr, "<span class='notice'>You think to yourself, \</span>"Hm.. this is only paper...\"")
+			log_admin("PAPER: [key_name(usr)] tried to use forbidden word in [src]: [bad].")
+			message_admins("PAPER: [key_name_admin(usr)] tried to use forbidden word in [src]: [bad].")
+			return
+*/
```

Comment:
```suggestion
```

Reply:
Didnt know if i should keep it or not, removing it.